app/models/manageiq/providers/google/inventory/parser.rb

```diff
@@ -338,15 +340,7 @@ def get_os_product_name(storage)
   # Get image's (miq_template's) ems_ref from
   # instance disks (connected to cloud_volume's `self_link`)
   def parse_instance_parent_image(instance)
-    parent_image_uid = nil
-
-    instance.disks.each do |disk|
-      parent_image_uid = @cloud_volume_url_to_source_image_id[disk[:source]]
-      next if parent_image_uid.nil?
-      break
-    end
-
-    parent_image_uid
+    instance.disks&.detect { |disk| @cloud_volume_url_to_source_image_id[disk[:source]] }
```

Comment:
Oops that's my bad it wasn't supposed to return the disk which is what detect is doing, how about

```suggestion
    instance.disks&.each do |disk|
      image_id = @cloud_volume_url_to_source_image_id[disk[:source]]
      return image_id if image_id
    end
    nil
```